### ISSUE TITLE
Analyzing implicits can confuse the compiler

### ISSUE THREAD

**Issue by raboof** [ensime/ensime-server#1176]
In the following snippet:

```scala
package org.example

import scala.concurrent.duration._
import scala.language.postfixOps

object Example {
  2 seconds
}
```

.. an implicit conversion from `scala.concurrent.duration.IntDuration` is used. This initially works, however after getting the implicits information for this while file (ImplicitInfoInRegion), suddenly this implicit is no longer applied.

**Comment by raboof** [comment]
I have added 2 reproduction tests in https://github.com/raboof/ensime-server/tree/sideEffectMarkingImplicits, one in `RichPresentationCompilerSpec` and one called ImplicitsBug.

From what I've been able to see so far it appears `RichPresentationCompiler.wrapLinkPos` appears to be the culprit and has an unexpected side-effect.

The test in `RichPresentationCompilerSpec` actually succeeds (in other words, fails to reproduce the problem). The seems to be because in this test the search service is not populated, causing `ModelBuilders._locateSymbolPos` not to call `askLinkPos` because it didn't find `scala.concurrent.duration.DurationInt`.

**Comment by fommil** [comment]
I want to close this as "can't reproduce" so much, but I have seen it with my own eyes. This is going to linger for years until some mad genius like @paulp comes along and  points at an obscure SI ticket from 2005 that hints at a terrifying inadequacy that makes us all question our reason for using this language.

**Comment by fommil** [comment]
Holy shit, you were able to reproduce? OK, all cards are being gathered again.

**Comment by aemoncannon** [comment]
Let's just restart the compiler on every request :-p

**Comment by fommil** [comment]
I'm going to have to call in the heavies: @sschaef @dragos have you ever seen anything like this?

**Comment by sschaef** [comment]
Indeed, I know this problem. ;) @dragos just fixed it a few week ago in Scala IDE. But I can't remember what the origin of the problem was. Give me some time, I'm going to search for the ticket/PR.

**Comment by sschaef** [comment]
I was mistaken - we couldn't fix the problem. Iulian only opened a ticket: https://scala-ide-portfolio.assembla.com/spaces/scala-ide/tickets/1002552-implicit-not-found-after-scaladoc-request/details#

I talked with Jason about it and we came to the conclusion that the problem probably arises because we don't use Scaladoc correctly in Scala IDE. I looked into that by myself but couldn't find the origin of the problem. @raboof reproduction case is similar to ours, is ensime also involved in retrieving information from Scaladoc?

**Comment by fommil** [comment]
interesting, I wouldn't have thought that scaladoc would play any part in this. We do have code that requests scaladoc information, but its not obvious that we'd be requesting that information here. CC @retronym because its starting to feel like a party in this ticket now.

**Comment by raboof** [comment]
My shot-in-the-dark guess (without looking at the code) would be that requesting scaladoc information, like our reproduction, calls `Global.getLinkPos` (or something even deeper) at some point.

**Comment by dragos** [comment]
That should be fixed in the upcoming 2.11.8. See scala/scala#4777

**Comment by fommil** [comment]
oh, wow! Can we try this out in a nightly release or something?

@raboof can you please submit the test (not the project test) as an ignored test?

**Comment by dragos** [comment]
Sure. Scala IDE nightlies should already have this fix.

**Comment by fommil** [comment]
ok, I guess we just need to figure out how to test against scala nightlies

**Comment by raboof** [comment]
@fommil unfortunately only the project test successfully reproduces the problem (the presentation compiler test still succeeds, I haven't been able to track down the difference yet).

I'll try to run the project test in https://github.com/raboof/ensime-server/tree/sideEffectMarkingImplicits against a Scala IDE nightly - unless someone beats me to it ;).

**Comment by fommil** [comment]
OK, commiting that as an ignored test would also be good.

**Comment by raboof** [comment]
Submitted as PR #1177

**Comment by hedefalk** [comment]
I hadn't seen this issue until it was referenced from ensime-atom by @joedirksen. https://github.com/ensime/ensime-atom/issues/116#issuecomment-185885328

So reason ensime-atom might have more problems is that we by default ask for implicits on the entirety of every file. That would make us hit these issues more often.

Seems @joedirksen also confirmed it was fixed upstream since compiling ensime-server against a 2.11.8-SNAPSHOT made it go away.

**Comment by hedefalk** [comment]
Oh, I see @raboof already confirmed it was fixed in 2.11.8. I gotta spend more time clicking all those reference links :)
